File: lib/models/producto.dart
class Producto {
  int id;
  int categoriaId;
  String nombre;
  String descripcion;
  double precio;
  int stock;
  String codigoBarras;
  String imagenUrl;
  bool activo;

  Producto({
    required this.id,
    required this.categoriaId,
    required this.nombre,
    required this.descripcion,
    required this.precio,
    required this.stock,
    required this.codigoBarras,
    required this.imagenUrl,
    required this.activo,
  });

  factory Producto.fromJson(Map<String, dynamic> json) => Producto(
        id: int.parse(json['id'].toString()),
        categoriaId: int.parse(json['categoria_id'].toString()),
        nombre: json['nombre'],
        descripcion: json['descripcion'] ?? '',
        precio: double.parse(json['precio'].toString()),
        stock: int.parse(json['stock'].toString()),
        codigoBarras: json['codigo_barras'] ?? '',
        imagenUrl: json['imagen_url'] ?? '',
        activo: json['activo'].toString() == '1',
      );
}

File: backend/models/Producto.php
<?php
class Producto {
    private $conn;
    private $table_name = "productos";

    public function __construct($db) {
        $this->conn = $db;
    }

    public function listar() {
        $query = "SELECT p.*, c.nombre AS categoria
                  FROM {$this->table_name} p
                  INNER JOIN categorias c ON p.categoria_id = c.id
                  ORDER BY p.id DESC";
        $stmt = $this->conn->prepare($query);
        $stmt->execute();
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    public function insertar($categoria_id, $nombre, $descripcion, $precio, $stock, $codigo_barras, $imagen_url, $activo) {
        $query = "INSERT INTO {$this->table_name}
            (categoria_id, nombre, descripcion, precio, stock, codigo_barras, imagen_url, activo)
            VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
        $stmt = $this->conn->prepare($query);
        return $stmt->execute([$categoria_id, $nombre, $descripcion, $precio, $stock, $codigo_barras, $imagen_url, $activo]);
    }

    public function actualizar($id, $categoria_id, $nombre, $descripcion, $precio, $stock, $codigo_barras, $imagen_url, $activo) {
        $query = "UPDATE {$this->table_name}
                  SET categoria_id=?, nombre=?, descripcion=?, precio=?, stock=?, codigo_barras=?, imagen_url=?, activo=?
                  WHERE id=?";
        $stmt = $this->conn->prepare($query);
        return $stmt->execute([$categoria_id, $nombre, $descripcion, $precio, $stock, $codigo_barras, $imagen_url, $activo, $id]);
    }

    public function eliminar($id) {
        $query = "DELETE FROM {$this->table_name} WHERE id=?";
        $stmt = $this->conn->prepare($query);
        return $stmt->execute([$id]);
    }
}
?>
